src/Vault.ts

```diff
@@ -0,0 +1,419 @@
+        investment
+      const supportsPermit = investmentCurrency.supportsPermit && 'send' in signer // eth-permit uses the deprecated send method
+      const needsApproval = investmentCurrencyAllowance.lt(amount)
+
+      if (!isAllowedToInvest) throw new Error('Not allowed to invest')
+      if (amount.gt(investmentCurrencyBalance)) throw new Error('Insufficient balance')
+      if (!amount.gt(0n)) throw new Error('Order amount must be greater than 0')
+
+      let permit: Permit | null = null
+      if (needsApproval) {
+        if (supportsPermit) {
+          permit = yield* doSignMessage('Sign Permit', () =>
+            signPermit(
+              walletClient,
+              signer,
+              self.chainId,
+              signingAddress,
+              investmentCurrency.address,
+              centrifugeRouter,
+              amount.toBigInt()
+            )
+          )
+        } else {
+          yield* doTransaction('Approve', publicClient, () =>
+            walletClient.writeContract({
+              address: investmentCurrency.address,
+              abi: ABI.Currency,
+              functionName: 'approve',
+              args: [centrifugeRouter, amount],
+            })
+          )
+        }
+      }
+
+      const enableData = encodeFunctionData({
+        abi: ABI.CentrifugeRouter,
+        functionName: 'enableLockDepositRequest',
+        args: [self.address, amount],
+      })
+      const requestData = encodeFunctionData({
+        abi: ABI.CentrifugeRouter,
+        functionName: 'executeLockedDepositRequest',
+        args: [self.address, signingAddress, estimate],
+      })
+      const permitData =
+        permit &&
+        encodeFunctionData({
+          abi: ABI.CentrifugeRouter,
+          functionName: 'permit',
+          args: [
+            investmentCurrency.address,
+            centrifugeRouter,
+            amount.toString(),
+            permit.deadline,
+            permit.v,
+            permit.r,
+            permit.s,
+          ],
+        })
+      yield* doTransaction('Invest', publicClient, () =>
+        walletClient.writeContract({
+          address: centrifugeRouter,
+          abi: ABI.CentrifugeRouter,
+          functionName: 'multicall',
+          args: [[enableData, requestData, permitData].filter(Boolean)],
+          value: estimate,
+        })
+      )
+    }, this.chainId)
+  }
+
+  /**
+   * Cancel an open investment order.
+   */
+  cancelInvestOrder() {
+    const self = this
+    return this._transactSequence(async function* ({ walletClient, signingAddress, publicClient }) {
+      const { centrifugeRouter } = lpConfig[self.chainId]!
+      const [estimate, investment] = await Promise.all([self.network._estimate(), self.investment(signingAddress)])
+
+      if (investment.pendingInvestCurrency.isZero()) throw new Error('No order to cancel')
+
+      yield* doTransaction('Cancel Invest Order', publicClient, () =>
```

Comment:
Will these tx descriptions be passed through and shown in the UI? If so, I would stick to our pattern of not capitalizing:
```suggestion
      yield* doTransaction('Cancel invest order', publicClient, () =>
```